FHICORC.Tests/BusinessRules/BorderControlRulesTests.cs

```diff
@@ -0,0 +1,547 @@
+            var results = ruleVerifierService.Verify(rules, rulesModel);
+            bool atLeastOneFalse = false;
+            foreach (var result in results)
+            {
+                if (result.Result == RulesFeedbackResult.FALSE)
+                {
+                    atLeastOneFalse = true;
+                    break;
+                }
+            }
+            Assert.AreEqual(true, atLeastOneFalse);
+        }
+
+        [Test]
+        public void Recovery_AfterValidPeriod_RedResult()
+        {
+            DCCPayload dccPayload = GetRecoveryPayload(RECOVERY_MIN_DAYS + 1, RECOVERY_MAX_DAYS, RECOVERY_MAX_DAYS);
+            VerifyRulesModel rulesModel = GetVerifyRulesModel(dccPayload);
+            rulesModel.External.ValidationClock = DateTime.UtcNow.AddDays(RECOVERY_MAX_DAYS);
+            var rules = ruleSelectorService.SelectRules(dccPayload);
+            var results = ruleVerifierService.Verify(rules, rulesModel);
+            foreach (var result in results)
+            {
+                Assert.AreEqual(RulesFeedbackResult.FALSE, result.Result);
+            }
+        }
+
+        [Test]
+        public void Recovery_InValidPeriod_UnknownType_RedResult()
+        {
+            DCCPayload dccPayload = GetRecoveryPayload(RECOVERY_MIN_DAYS, 0, RECOVERY_MAX_DAYS);
+            dccPayload.DCCPayloadData.DCC.Recovery[0].Disease = "UNKNOWN";
+            VerifyRulesModel rulesModel = GetVerifyRulesModel(dccPayload);
+            var rules = ruleSelectorService.SelectRules(dccPayload);
+            var results = ruleVerifierService.Verify(rules, rulesModel);
+            bool atLeastOneFalse = false;
+            foreach (var result in results)
+            {
+                if (result.Result == RulesFeedbackResult.FALSE)
+                {
+                    atLeastOneFalse = true;
+                    break;
+                }
+            }
+            Assert.AreEqual(true, atLeastOneFalse);
+        }
+
+        [Test]
+        public void Recovery_BeforeValidPeriod_UnknownType_RedResult()
+        {
+            DCCPayload dccPayload = GetRecoveryPayload(RECOVERY_MIN_DAYS, 0, RECOVERY_MAX_DAYS);
+            dccPayload.DCCPayloadData.DCC.Recovery[0].Disease = "UNKNOWN";
+            VerifyRulesModel rulesModel = GetVerifyRulesModel(dccPayload);
+            rulesModel.External.ValidationClock = DateTime.UtcNow.AddDays(-1);
+            var rules = ruleSelectorService.SelectRules(dccPayload);
+            var results = ruleVerifierService.Verify(rules, rulesModel);
+            bool atLeastOneFalse = false;
+            foreach (var result in results)
+            {
+                if (result.Result == RulesFeedbackResult.FALSE)
+                {
+                    atLeastOneFalse = true;
+                    break;
+                }
+            }
+            Assert.AreEqual(true, atLeastOneFalse);
+        }
+
+        [Test]
+        public void Recovery_AfterValidPeriod_UnknownType_RedResult()
```

Comment:
We do not get the type of test used for recovery, it is not part of the data. We should only test on the period. Change name of test to reflect this.

Reply:
This is covered by another test then. Deleted this one